### ISSUE TITLE
Regression: specifiying a nic creates a configuration that no longer fetches any IP

### ISSUE THREAD

**Issue by danimo** [OpenNebula/terraform-provider-opennebula#293]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.2

### Affected Resource(s)

- opennebula_virtual_machine

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
```

### Expected Behavior

Providing a `nic {}` section in a `opennebula_virtual_machine` resource brings up a NIC with IP adesses from the Networks pool

### Actual Behavior

Providing a `nic {}` section in a `opennebula_virtual_machine` resource provides an unconfigured NIC

### Steps to Reproduce

1. Create a trivial `opennebula_virtual_machine` resource
2. Create a valid `nic {}` section like so:

```terraform
  nic {
    network_id  = 1
    security_groups = [0]
  }
```
3. `terraform apply`

**Results**

* Test with `0.4.3` => works as describe in `Expected Behavior`
* Upgrade to `0.5.0` =>  fails as described in `Actual Behavior`

Bisecting suggests that the breaking commit is e5c60343eb4bc9f7cf0d3084db5fc0b41e12255c.

**Comment by treywelsh** [comment]
Thanks for this detailed report. It's a duplicated of #289 and it seems that we agree on the faulty commit.
However I'm surprised that CI acceptance tests didn't fails....

I close this one to track the problem in #289

**Comment by frousselet** [comment]
Issue should be addressed in 0.5.1 (released early July).

**Comment by danimo** [comment]
Thanks, sorry for not spotting https://github.com/OpenNebula/terraform-provider-opennebula/issues/289 erlier.
